### ISSUE TITLE
cy.wait does not return api json when statusCode is 304

### ISSUE THREAD

**Issue by jarretmoses** [cypress-io/cypress#14277]
### Current behavior

I've currently upgraded to 6.2.0 and have updated my .route calls to .intercept. I have a test suite that makes an identical api call per test block. I've noticed that on the second test I get a failure as it is relying on the return data from my api but the statusCode is 304 and the body in the response becomes an empty string. Running the test locally and forcing the second test to run alone (.only) the response is 200 and the json returns as expected.
### Desired behavior
I believe the json from an original call to a api (ie NOT stubbed response) should return if the status code is 304.

### Test code to reproduce

Apologies I will work to create a full backend + frontend setup for this but may take a little time. I want to at least flag it to know 1 if this is a duplicate I missed or 2 if this is expected behavior and I missed something in the documentation. The High level setup is I have a backend, added an intercept such as `cy.intercept('GET', '/api/v1/buildings*').as('getBuildings');` and then in 2 subsequent tests used `cy.wait('@getBuildings').then` the second test has the 304 and thus fails (but passed when I ran it alone as it had status code 200)

### Versions

Current Version: 6.2.0
Previous Working Version: 5.x

**Comment by elihschiff** [comment]
I was having what I think was the same issue, and I found a solution that might work for you as well. https://glebbahmutov.com/blog/cypress-intercept-problems/#the-solution

Pretty much you need to change 
```js
cy.intercept('GET', '/api/v1/buildings*').as('getBuildings');
```
to 
```js
cy.intercept('GET', '/api/v1/buildings*', req => {
     delete req.headers['if-none-match']
}).as('getBuildings');
```
because this stops the browser from using a cached result

**Comment by joszo** [comment]
I'm facing simillar issue, I have 2 tests. If a run test 1 and then test2 there is statusCode:304 but in devTools my request has statusCode 200.

**Comment by jarretmoses** [comment]
the header deletion seems to work. If this is expected behavior then i will close this issue

**Comment by flotwig** [comment]
@jarretmoses hmm, this should have been fixed with 6.1.0: #8934

We are trying to make working with the cache using `cy.intercept` as intuitive as possible, so waiting on 304's should be successful. But deleting cache control headers certainly avoids bugs like this :laughing: 

If you have a repro for this bug that works in 6.2.0 or newer, I'd love to take a look. Otherwise, if your issue is fixed, we can go ahead and close this.

**Comment by flotwig** [comment]
Closing since no response - this should be fixed in 6.2.0+. If you're still experiencing this, please open a new issue.
